serving/docs/lmi_new/deployment_guide/benchmarking-your-endpoint.md

```diff
@@ -1,5 +1,108 @@
+This document will focus on how to benchmark your sagemaker endpoint with a lightweight tool, `awscurl`.
+
+Within LMI, we use a lightweight tool [`awscurl`](https://github.com/deepjavalibrary/djl-serving/tree/master/awscurl) for benchmarking.
+`awscurl` is a tool that provides a `curl` like API to make requests to AWS Services. It provides the following features:
+
+* Ability to launch concurrent clients 
+* Ability to specify number of requests per client
+* Ability to specify a tokenizer to get reliable token counts
+* Ability to provide a list of request bodies to benchmark your endpoint across a variety of input/output lengths, and generation parameters
+* Reports metrics like:
+  * Total Time taken for benchmark
+  * Non 200 responses and error rate
+  * Average, P50, P90, P99 latency
+  * Requests Per second (TPS)
+  * Tokens per Request (generated tokens)
+  * Tokens per Second (total Throughput of system across all clients/requests)
+  * Time to first Byte (for streaming, represents the time to first token)
+
+We will now walk through some examples of using `awscurl` to benchmark your SageMaker Endpoint.
+
+## awscurl Usage
+
+To install awscurl, and setup AWS credentials, see the `awscurl` [documentation](https://github.com/deepjavalibrary/djl-serving/tree/master/awscurl).
+
+To see a list of available options, run `./awscurl -h`.
+
+The options we use in the examples below are:
+
+* `-c`: number of concurrent clients
+* `-N`: number of requests per client (c * N is the total number of requests made during the benchmark)
+* `-n`: name of aws service to call (will always be `sagemaker` for this use-case)
+* `-X`: HTTP method to use (will always be `POST` for this use-case)
+* `--connect-timeout`: time in seconds to wait for response (sagemaker has default 60 second invocation time, so typically this is 60)
+* `-d`: request body JSON (this is a single request body that will be used across all clients and requests)
+* `--dataset`: a path to directory of files, or path to a single file that contain payloads 
+* `-P`: print output in JSON
+* `-t`: report tokens per second in benchmark metrics
+* `-H`: custom HTTP Headers
+* `-o`: output file prefix to save results of requests (1 file is generated per client)
+
+Additionally, we recommend that you set the `TOKENIZER` environment variable to the value of your model's tokenizer (either HuggingFace Hub model id, or local path where the `tokenizer.json` file is present).
+If a tokenizer is not specified, and tokens per second is requested, `awscurl` will use the number of words in the response.
+
+### Usage with a single common payload
+
+To run a benchmark with 10 concurrent clients, each issuing 30 requests, with a single common payload, we can run:
+
+```shell
+TOKENIZER=<tokenizer_id> ./awscurl -c 10 -N 30 -X POST -n sagemaker <your full sagemaker endpoint> (e.g. https://runtime.sagemaker.us-east-1.amazonaws.com/endpoints/mms-demo/invocations) \
```

Comment:
Better add example for benchmarking Time to first byte using `invocations-response-stream` endpoint on SM

Reply:
Will add that example

Reply:
added section on streaming usage (does depend on new awscurl being published to s3 to fix the issue)